api/catalog/api/controllers/elasticsearch/search.py

```diff
@@ -0,0 +1,198 @@
+            "extension",
+            "category",
+            ("categories", "category"),
+            "aspect_ratio",
+            "size",
+            "length",
+            "source",
+            ("license", "license.keyword"),
+            ("license_type", "license.keyword"),
+        ],
+        "exclude": [
+            ("excluded_source", "source"),
+        ],
+    }
+    for behaviour, bases in rules.items():
+        for basis in bases:
+            s = _apply_filter(s, query_serializer, basis, behaviour)
+
+    # Exclude mature content
+    if not search_params["mature"]:
+        s = s.exclude("term", mature=True)
+    # Exclude sources with ``filter_content`` enabled
+    s = _exclude_filtered_providers(s)
+
+    # Search either by generic multimatch or by "advanced search" with
+    # individual field-level queries specified.
+
+    search_fields = ["tags.name", "title", "description"]
+    if "q" in search_params:
+        query = _quote_escape(search_params["q"])
+        s = s.query(
+            "simple_query_string",
+            query=query,
+            fields=search_fields,
+            default_operator="AND",
+        )
+        # Boost exact matches
+        quotes_stripped = query.replace('"', "")
+        exact_match_boost = Q(
+            "simple_query_string",
+            fields=["title"],
+            query=f'"{quotes_stripped}"',
+            boost=10000,
+        )
+        s.query = Q("bool", must=s.query, should=exact_match_boost)
+    else:
+        query_bases = ["creator", "title", ("tags", "tags.name")]
+        for query_basis in query_bases:
+            if isinstance(query_basis, tuple):
+                serializer_field, es_field = query_basis
+            else:
+                serializer_field = es_field = query_basis
+            if serializer_field in search_params:
+                value = _quote_escape(search_params[serializer_field])
+                s = s.query("simple_query_string", fields=[es_field], query=value)
+
+    if settings.USE_RANK_FEATURES:
+        feature_boost = {"standardized_popularity": 10000}
+        rank_queries = []
+        for field, boost in feature_boost.items():
+            rank_queries.append(Q("rank_feature", field=field, boost=boost))
+        s.query = Q("bool", must=s.query, should=rank_queries)
+
+    # Use highlighting to determine which fields contribute to the selection of
+    # top results.
+    s = s.highlight(*search_fields)
+    s = s.highlight_options(order="score")
+
+    # Route users to the same Elasticsearch worker node to reduce
+    # pagination inconsistencies and increase cache hits.
+    s = s.params(preference=str(ip), request_timeout=7)
+
+    # Paginate
+    start, end = _get_query_slice(
+        s,
+        search_params["page_size"],
+        search_params["page"],
+        search_params["filter_dead"],
+    )
+    s = s[start:end]
+
+    try:
+        if settings.VERBOSE_ES_RESPONSE:
+            log.info(pprint.pprint(s.to_dict()))
```

Comment:
I don't know if we have an issue open for this already, but this is great! It'll surely reduce the noise in our logs 💯

Reply:
This already existed before, it's not new or added as a part of this PR.

Reply:
Oh 😂